src/Libraries/CoreNodeModelsWpf/Charts/Controls/BarChartControl.xaml.cs

```diff
@@ -67,81 +67,97 @@ private void NodeModel_PropertyChanged(object sender, PropertyChangedEventArgs e
             {
                 var model = sender as BarChartNodeModel;
 
-                // Invoke on UI thread
-                this.Dispatcher.Invoke(() =>
+                try
                 {
-                    BarChart.Series.Clear();
-
-                    // Load sample data if any ports are not connected
-                    if (!model.InPorts[0].IsConnected && !model.InPorts[1].IsConnected && !model.InPorts[2].IsConnected)
-                    {
-                        DefaultSeries();
-                    }
-                    else
+                    // Invoke on UI thread
+                    this.Dispatcher.Invoke(() =>
                     {
-                        var seriesRange = UpdateSeries(model);
-
-                        BarChart.Series.AddRange(seriesRange.ToArray());
-                    }
-                });
+                        BarChartNode.Series = Enumerable.Empty<ISeries>();
+
+                        // Load sample data if any ports are not connected
+                        if (!model.InPorts[0].IsConnected && !model.InPorts[1].IsConnected && !model.InPorts[2].IsConnected)
+                        {
+                            DefaultSeries();
+                        }
+                        else
+                        {
+                            var seriesRange = UpdateSeries(model);
+                            BarChartNode.Series = BarChartNode.Series.Concat(seriesRange);
+                        }
+                    });
+                }
+                catch(Exception ex) {
+                    Console.WriteLine(ex.ToString());
+                    Console.WriteLine(ex.Message.ToString());
```

Comment:
I am interested in the exception handling here, in your opinion, what could be the exception here, and is it worth surfacing it to the user as warning?

Reply:
will remove, I had these for some tests during development

Reply:
Not removed yet ;)

Reply:
sorry I missed it, thanks for noticing, removed now :)